doc/source/guide/plotting.rst

```diff
@@ -127,65 +127,126 @@ plot window blocks the command line terminal and must be closed before doing any
+contours on the Map:
 
 .. plot::
     :include-source:
 
+    import matplotlib.pyplot as plt
+    import astropy.units as u
+
     import sunpy.map
     import sunpy.data.sample
-    import matplotlib.pyplot as plt
-    from astropy.units import Quantity
-    smap = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
-    smap.plot()
-    smap.draw_limb()
+
+    aia_map = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+    aia_map.plot()
+    aia_map.draw_limb()
+
     # let's add contours as well
-    smap.draw_contours(Quantity([10,20,30,40,50,60,70,80,90], 'percent'))
+    aia_map.draw_contours([10,20,30,40,50,60,70,80,90] * u.percent)
+
     plt.colorbar()
     plt.show()
 
-For more advanced plotting you'll want to create the `~matplotlib.figure.Figure` object yourself.
-The following example plot shows how to add a rectangle to a plot to, for example,
-highlight a region of interest, and change the plot title.
+
+In this example, the `~matplotlib.figure.Figure` and `~wcsaxes.WCSAxes`
+instances are created explicitly, and then used to modify the plot:
 
 .. plot::
     :include-source:
 
+    import matplotlib.pyplot as plt
+    import astropy.units as u
+
     import sunpy.map
     import sunpy.data.sample
-    import matplotlib.pyplot as plt
-    from matplotlib import patches
+
     smap = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
 
     fig = plt.figure()
-    ax = plt.subplot()
+    # Provide the Map as a projection, which creates a WCSAxes object
+    ax = plt.subplot(projection=smap)
 
-    smap.plot()
-    rect = patches.Rectangle([-350, -650], 500, 500, color = 'white', fill=False)
+    im = smap.plot()
+
+    # Prevent the image from being re-scaled while overplotting.
+    ax.set_autoscale_on(False)
+
+    xc = [0,100,1000] * u.arcsec
+    yc = [0,100,1000] * u.arcsec
+
+    # ax.get_transform() tells WCSAxes what coordinates to plot in.
+    # 'world' coordinates are always in degrees
+    p = plt.plot(xc.to(u.deg), yc.to(u.deg), 'o',
+                 transform=ax.get_transform('world'))
+
+    # Set title.
     ax.set_title('My customized plot')
-    ax.add_artist(rect)
+    
     plt.colorbar()
     plt.show()
 
+It is possible to create the same plot, explicitly not using `wcsaxes`, however
+this example will not have the features of `wcsaxes` which include correct
+representation of rotation and plotting in different coordinate systems.
+
+.. plot::
+    :include-source:
+
+    import matplotlib.pyplot as plt
+    import astropy.units as u
+
+    import sunpy.map
+    import sunpy.data.sample
+
+    smap = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
+
+    fig, ax = plt.subplots()
+
+    im = smap.plot()
+
+    # Prevent the image from being re-scaled while overplotting.
+    ax.set_autoscale_on(False)
+
+    xc = [0,100,1000] * u.arcsec
+    yc = [0,100,1000] * u.arcsec
+
+    p = plt.plot(xc, yc, 'o')
+
+    # Set title.
+    ax.set_title('My customized plot')
```

Comment:
Maybe add in the title : "with/without wcsaxes" will help to differentiate the output of both.